File: AngularBooking/Models/Venue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AngularBooking.Models
{
    public class Venue : IModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public string Address1 { get; set; }
        public string Address2 { get; set; }
        public string Address3 { get; set; }
        public string Address4 { get; set; }
        public string Address5 { get; set; }
        public string ContactPhone { get; set; }
        public string LatLong { get; set; }
        public string Website { get; set; }
        public string Facebook { get; set; }
        public string Twitter { get; set; }
        public string Instagram { get; set; }
        public FacilityFlags Facilities { get; set; }

        // navigation
        public List<Room> Rooms { get; set; }
    }
}

File: AngularBooking/ClientApp/src/app/core/_models/entity/venue.ts
import { FacilityFlags } from "./facility-flags";
import { Room } from "./room";

export class Venue {
  constructor(
    public id?: number,
    public name?: string,
    public description?: string,
    public image?: string,
    public address1?: string,
    public address2?: string,
    public address3?: string,
    public address4?: string,
    public address5?: string,
    public contactPhone?: string,
    public latLong?: string,
    public website?: string,
    public facebook?: string,
    public twitter?: string,
    public instagram?: string,
    public facilities?: FacilityFlags,
    public rooms?: Room[]
  ) { }
}
